vpd-tool/include/tool_utils.hpp

```diff
@@ -125,5 +126,92 @@ inline std::string getPrintableValue(const types::BinaryVector& i_keywordValue)
 
     return l_oss.str();
 }
+
+/**
+ * @brief API to save keyword's value on file.
+ *
+ * API writes keyword's value on the given file path. If the data is in hex
+ * format, API strips '0x' and saves the value on the given file.
+ *
+ * @param[in] i_filePath - File path.
+ * @param[in] i_keywordValue - Keyword's value.
+ *
+ * @return - true on successfully writing to file, false otherwise.
+ */
+inline bool saveToFile(const std::string& i_filePath,
+                       const std::string& i_keywordValue)
+{
+    bool l_returnStatus = false;
+
+    if (i_keywordValue.empty())
+    {
+        // ToDo: log only when verbose is enabled
+        std::cerr << "Save to file[ " << i_filePath
+                  << "] failed, reason: Empty keyword's value received"
+                  << std::endl;
+        return l_returnStatus;
+    }
+
+    std::string l_keywordValue{i_keywordValue};
+    if (i_keywordValue.substr(0, 2).compare("0x") == constants::STR_CMP_SUCCESS)
+    {
+        l_keywordValue = i_keywordValue.substr(2);
+    }
+
+    std::ofstream l_outPutFileStream;
+    try
+    {
+        l_outPutFileStream.open(i_filePath);
+
+        if (l_outPutFileStream.is_open())
+        {
+            l_outPutFileStream.exceptions(std::ifstream::badbit |
```

Comment:
keeping this here instead before calling open(), 
If we set early it throws as error, if folder path not there with generic error "basic_ios::clear iostream error".
Instead is_open() will get fail & will display "Error opening output file" log (as current behaviour).

and any error in write() now will be get caught in catch().

Reply:
Now I am confused on what will happen

Reply:
just move this outside the try . any issues in that